HO adding all article files. (#149)
hassan-obeid/tr_b_causal_2020: article/sections/_7_latent_confounding.tex

\section{Latent Confounding}
\label{sec:latent-confounding}

This section focuses on the more complicated yet realistic case that is 
typically faced by demand modelers, where we have latent confounders that 

> affect the treatment assignment of our causal variables of interest, as well as the outcome. 

Reviewer: ```suggestion
This section focuses on the more complicated yet realistic case that is 
typically faced by demand modelers, where we latent confounders
affect the treatment assignment of the causal variables of interest, as well as the outcome variables. 
```

> We first go over a few examples of confounding in transportation analyses, explain 
> the challenges that come with such cases, and present a few approaches to 
> dealing with this issue, specifically the recent de-confounder technique of 
> \citet{wang_2019_blessings}. 

Reviewer: ```suggestion
We first present some examples of latent confounding in transportation analyses, explain 
the challenges that come with such cases, and present a few approaches to 
dealing with this issue, most notably the recent de-confounder technique by
\citet{wang_2019_blessings}. 
```

Our application will show how directed acyclic graphs can 
help increase transparency about one's reasoning regarding the number of 
confounders, assumptions for which variables are confounded, and the models 

> needed to estimate the causal effects. 

Reviewer: Should we use DAGs and causal graphs interchangeably throughout the report or should we stick with one?

```suggestion
Our application will show how DAGs can make one's reasoning on the number of confounders, assumptions for which variables are confounded, and the models needed to estimate the causal effects more transparent. 
```

In addition to the application above, we use simplified simulation scenarios to further investigate the usefulness 

> and pitfalls of this approach for generating accurate model estimates. 

Reviewer: ```suggestion
We also use simplified simulation scenarios to further investigate the usefulness and pitfalls of the de-confounder in generating accurate model estimates. 
```

\subsection{Examples of confounding}
\label{sec:confounding-examples}

> Confounding occurs when a certain (confounding) variable induces variations in
> both the outcome as well as the treatment (policy) variables of interest, 
> creating correlation between the treatment and the outcome that is not caused
> by the treatment variable itself. When the confounding variable (or variables) is 
> observed, we can control for the confounding effect, and there exists many 
> methods in the literature for how to do that, including post-stratification,
> multiple regression, propensity score methods, etc. It is when the 
> confounding variable is unobserved that the problem becomes significantly 
> more challenging. 

Reviewer: ```suggestion
Confounding occurs when a certain variable induces variations in
both the outcome as well as the treatment variables of interest, 
creating correlation between the treatment and the outcome that is not caused
by the treatment variable itself. When the confounding variables are 
observed, one should control for the confounding effect. There are many 
methods in the literature for doing just that, including post-stratification,
multiple regression, propensity score methods, etc. However, when confounding variables are unobserved, controlling for them becomes significantly more challenging.
```

> For an illustration, imagine we're interested in the effect of adding a bike on the mode share of bicycling in a given neighborhood. 

Reviewer: ```suggestion
As an illustration of such difficulty, imagine we are interested in estimating the effect of adding a bike lane on the mode share of bicycling in a given neighborhood. 
```

> To estimate this effect, one strategy that a modeler may use is to develop a disaggregate mode choice model with a dummy variable for whether a bikelane exists between an individual's trip origin and destination, and then treating the coefficient on this variable as the causal effect of adding a bikelane on the log-odds of choosing to bike. 

Reviewer: ```suggestion
To estimate this effect, one strategy that a modeler may use is to develop a disaggregate mode choice model with a dummy variable for whether a bike lane exists between an individual's trip origin and destination. The modeler would then treat the coefficient on this variable as the causal effect of adding a bike lane on the log-odds of choosing to bike. 
```

> The problem with such strategy is that individuals may self-select to live in an area where bicycle infrastructure exists because they have a preference for commuting by bike. 

Reviewer: ```suggestion
One problem with this strategy is that it fails to considers the problem of self-selection into treatment; Individuals may self-select to live in an area where bicycle infrastructure exists because they have a preference for commuting by bike. 
```

> In other words, if we define an additional variable to encode a person's latent inherent preference for biking, then this variable determines both a person's likelihood for living in an area with an existing bicycle infrastructure, as well as whether that person choose to bike. 

Reviewer: ```suggestion
In other words, if we define an additional variable to encode a person's latent inherent preference for biking, then this variable determines both a person's likelihood for living in an area with an existing bicycle infrastructure, as well as whether that person choose to commute by bike. 
```

This variable is subsequently expected not to be "balanced" across the treatment and control groups; those who live near a bicycle lane are expected to have higher preference for biking than those who don't. 

> Without accounting for this latent confounder, we risk getting very biased estimates 
> of the effects of our treatment variable of interest, created by the 
> induced variations by the confounder in both the treatment and the outcome 
> variable. 

Reviewer: Removed "very" because I don't think that the degree of bias matters, or does it?

```suggestion
Without accounting for this latent confounder, the analyst risks getting biased estimates 
of the effects of the treatment variable of interest, created by the 
induced variations by the confounder in both the treatment and the outcome 
variable. 
```

> The problem of latent confounding and ommitted variable bias is widely 
> acknowledged in the transportation literature, and demand modelers can draw on 
> a list of method to account for confouding in some specific circumstances. 

Reviewer: ```suggestion
The problem of latent confounding and ommitted variable bias is widely 
acknowledged in the transportation literature. Travel demand modelers can draw on 
a list of methods to account for confouding in some specific circumstances. 
```

> Integrated choice and latent variable (ICLV) models are a way to account for 
> the effect of unobserved attitudinal variables which may affect the selection 
> of some individuals into some treatment level, as well as an outcome of 

Reviewer: ```suggestion
Integrated choice and latent variable (ICLV) models help account for 
the effect of unobserved attitudinal variables which may affect the selection 
of some individuals into some treatment level and the outcome of 
```

> interest. For example, a person's unobserved beliefs and attitudes towards 
> being environmentelly friendly may both affect whether she chooses to bike to 
> work, as well as whether she lives close to a bike infrastructure, which may 
> bias our observational study if we're interested in the effect adding a 
> bikelane has on the mode share of bicycles. Those methods, however, rely on 
> collecting attitudinal indicators typically obtained by conducting
> additional surveys with a sample of the people. This may not be an option in 
> many cases; for example, we may not be able to reach the individuals to conduct additional surveys with them. 

Reviewer: ```suggestion
interest. For example, a person's unobserved beliefs and attitudes towards 
transportation's environmental impact may affect whether she chooses to bike to 
work and whether she lives close to a bike infrastructure. This may in turn 
bias our observational study if we are interested in the effect that adding a 
bike lane has on the mode share of bicycles. ICLV methods, however, rely on 
collecting attitudinal indicators typically obtained by conducting
additional surveys with a sample of the people. This may not be an option in 
many cases; for example, we may not be able to reach the individuals to conduct additional surveys with them. 
```

\subsection{The deconfounder algorithm}
\label{sec:deconfounder-algo}

One recent method that has been proposed to deal with the problem of latent 
confounding without the need to collect additional data is the deconfounder 
algorithm by \citet{wang_2019_blessings}. The method attempts to control for the 
confouding variable by estimating a "subsitute confounder", a set of variables 
that once controlled for, renders all variation in the treatment variables of 
interest exogenous. The process of applying the method is actually quite 

> straightforward and simple. It proceeds as follows:

Reviewer: ```suggestion
\citet{wang_2019_blessings} recently proposed the deconfounder algorithm as a method to deal with latent confounding without needing to collect additional data. The deconfounder attempts to control for the 
confouding variable by estimating a "substitute confounder", a set of variables 
that once controlled for, render all variation in the treatment variables of 
interest exogenous. The process of applying the method is
straightforward and simple. It proceeds as follows:
```

\begin{itemize}
	\item First, estimate the substitute confounder using any good latent variable 
	model the modeler chooses. The authors suggest estimating a factor model 
	with k factors on the set of covariates the modeler is interested in. 
	\item Second, check the factor model's accuracy using posterior predictive 
	checks (add details). 
	\item Once a sufficiently accurate latent variable model is recovered, use it 
	to estimate an expected value of the latent variable for each observation, 
	and control for this value in the outcome model, alongside the treatment 
	variables and other covariates of interest. 
\end{itemize}

One main assumption of the deconfounder algorithm is that the data at hand 
should have multi-cause confounders, thus the title of the paper, "The 
blessings of multiple causes". In other words, this method works when the 
unobserved confounders affect multiple of the observed causes (or treatment 
variables) of interest, alongside the outcome. This assumption is weaker than 
the one required for ignorability to hold, which requires the absence of both 

> single cause and multi-cause confounders for accurate causal inferences. 

Reviewer: I'm confused by the last sentence in this paragraph. Are we saying that this assumption is hard to hold?

```suggestion
One main assumption of the deconfounder algorithm is that the data at hand 
should have multi-cause confounders. In other words, the deconfounder algorithm works when the 
unobserved confounders affect many observed causes (or treatment 
variables) of interest, alongside the outcome. This assumption is weaker than 
the one required for ignorability to hold, which requires the absence of both 
single cause and multi-cause confounders for accurate causal inferences. 
```

Reviewer: I would also say in the last sentence in this section what the next subsection is about and delete the first sentence from the next subsection. Just a suggestion.

\subsection{Case study: Simulation}
\label{sec:deconfounder-simulation}

The purpose of this section is to investigate the effectiveness of the 
deconfounder algorithm \citep{wang_2019_blessings} in adjusting for unobserved 
confounding. We use a simulated mode choice data where travel distance 
linearly confounds both travel time and travel cost. We then mask the travel 

> distance data and treat it as an unobserved variable. 

Reviewer: This is the same dataset used in section 2. Should we make that clear here?

	where a modeler ignores, or is unaware of, unobserved confounding.
	\item Model 2: We use the deconfounder algorithm to try to recover the 
	confounder (travel distance). In this method, we use all the variables in 
	each mode's utility to recover that mode's confounder. This is in line 
	with the approach taken in \citet{wang_2019_blessings}, where they use all the set 
	of observed variables in the factor model to recover a substitute 
	confounder.
	\item Model 3: We use the deconfounder algorithm to try to recover the 
	confounder (travel distance), but this time, we only use travel time and 
	cost in the factor model, instead of all the variables in the utility 
	specification of each mode. By only using what we know are confounded 
	variables to recover the substitute confounder, our goal is to analyze 
	whether we can improve the accuracy of this approach by adopting a more 
	thoughtful approach to hypothesizing on the confounder variables. This can 
	be in the form of building and testing candidate causal graphs that 
	try to illustrate this confounding. 
\end{itemize}

We compare both the estimates of the coefficients on travel time and cost from 

> each of those three models to the true estimates used in the simulation, as well as the distribution of the recovered substitute confounder under each of 

Reviewer: ```suggestion
We compare the estimates of the coefficients on travel time and travel cost from 
each of the three described models to the true estimates used in the simulation, as well as the distribution of the recovered substitute confounder under each of 
```

models 2 and 3 to the true confounder. The main findings of this exercise are the following:

Using the true variables believed to be confounded (i.e. method 3 where only 
travel time and cost are used to recover the confounder) leads to a better 
recovery of the true confounder. Figure \ref{fig:qq-plot-method-2} and \ref{fig:qq-plot-method-3} show a QQ plot of the true 

> and recovered confounders under models 2 and 3 respectively. Looking at those 
> figures, one can see that the distribution of the recovered substitute 

Reviewer: ```suggestion
and recovered confounders under models 2 and 3 respectively. Both figures show that the distribution of the recovered substitute 
```

confounder under method 3 is closer to that of the true confounder than method 
2. This suggests that it may be better to run the deconfounder algorithm based 
on a hypothesized causal graph, rather than just running it on all the 

> observed covariates. Please refer to Section \ref{sec:graph-construction} for how to build plausible causal graphs. 

Reviewer: I'm a little confused by this. Do you mean it might be better to run it on covariates of interest given a certain causal graph?

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{qq-plot-method-2.png}
   \caption{QQ plot of true confounder against recovered confounder using all covariates}
   \label{fig:qq-plot-method-2}
\end{figure}

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{qq-plot-method-3.png}
   \caption{QQ plot of true confounder against recovered confounder using all covariates}
   \label{fig:qq-plot-method-3}
\end{figure}

% \begin{figure}
% \centering
% \begin{minipage}{.5\textwidth}
%   \centering
%   \includegraphics[width=.4\linewidth]{qq-plot-method-2.png}
%   \captionof{figure}{QQ plot of true confounder against recovered confounder using all covariates}
%   \label{fig:qq-plot-method-2}
% \end{minipage}%
% \begin{minipage}{.5\textwidth}
%   \centering
%   \includegraphics[width=.4\linewidth]{qq-plot-method-3.png}
%   \captionof{figure}{QQ plot of true confounder against recovered confounder using only confounded covariates}
%   \label{fig:qq-plot-method-3}
% \end{minipage}
% \end{figure}

> Additionally, and perhaps most importantly, the effectiveness of the 

Reviewer: ```suggestion
More importantly, we should note that the effectiveness of the 
```

deconfounder algorithm is very sensitive to small errors and misfits in the 

> recovered confounder. Although method 3 returns a relatively good fit of the 

Reviewer: ```suggestion
recovered confounder. Although model 3 returns a relatively good fit of the 
```
